src/useq/_plate.py

```diff
@@ -165,7 +186,9 @@ def _validate_well_points_plan(
     ) -> Any:
         value = handler(value)
         if plate := info.data.get("plate"):
-            if isinstance(value, RandomPoints):
+            if isinstance(value, RandomPoints) and (
+                value.max_width == np.inf or value.max_height == np.inf
```

Comment:
maybe do this on a per-field basis?  this will overwrite both of them if either of them is missing.  Also, could allow shape to pass through if provided (I believe you can check that with `'shape' not in value.__pydantic_fields_set__`

Reply:
@tlambert03 do you mean something like that?

```py
@field_validator("well_points_plan", mode="wrap")
@classmethod
def _validate_well_points_plan(
    cls,
    value: Any,
    handler: core_schema.ValidatorFunctionWrapHandler,
    info: core_schema.ValidationInfo,
) -> Any:
    value = handler(value)
    if plate := info.data.get("plate"):
        if isinstance(value, RandomPoints):
            plate = cast(WellPlate, plate)
            kwargs = value.model_dump(mode="python")
            if value.max_width == np.inf:
                kwargs["max_width"] = plate.well_size[0] - (value.fov_width or 0.1)
            if value.max_height == np.inf:
                kwargs["max_height"] = plate.well_size[1] - (value.fov_height or 0.1)
            if 'shape' not in value.__pydantic_fields_set__:
                kwargs["shape"] = (
                    Shape.ELLIPSE if plate.circular_wells else Shape.RECTANGLE
                )
            value = RandomPoints(**kwargs)
    return value
```

Reply:
i do!